src/Shell/Task/ModelTask.php

```diff
@@ -395,17 +441,74 @@ public function findBelongsToMany($model, array $associations)
                 $assocTable = substr($otherTableName, 0, $otherOffset);
             }
             if ($assocTable && in_array($assocTable, $tables)) {
-                $habtmName = $this->_camelize($assocTable);
-                $assoc = [
-                    'alias' => $habtmName,
-                    'foreignKey' => $foreignKey,
-                    'targetForeignKey' => $this->_modelKey($habtmName),
-                    'joinTable' => $otherTableName
-                ];
-                if ($assoc && $this->plugin) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+                if (!in_array($otherTableName, $tables)) {
+                    continue;
+                }
+                $assocTableObject = TableRegistry::get($otherTableName);
+
+                $thisForeignKey = $primaryKey;
+                foreach ($thisForeignKey as $i => $fieldName) {
+                    if ($fieldName === 'id') {
+                        $thisForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
+                    }
+                }
+                $otherForeignKeys = $assocTableObject->schema()->columns();
+                foreach ($otherForeignKeys as $i => $fieldName) {
+                    if ($fieldName !== 'parent_id' && (
+                        strlen($fieldName) <= 3 || !preg_match('/^.*_id$/', $fieldName))
+                    ) {
+                        unset($otherForeignKeys[$i]);
+                    }
+                }
+                $thisForeignKey = array_values($thisForeignKey);
+                $otherForeignKeys = array_values($otherForeignKeys);
+                $thisForeignKeyIntersection = array_intersect($thisForeignKey, $otherForeignKeys);
+
+                $reverseForeignKey = false;
+                $reverseForeignKeyIntersection = false;
+                $otherForeignKeysDiff = array_diff($otherForeignKeys, $thisForeignKey);
+                foreach ($otherForeignKeysDiff as $reverseForeignKey) {
```

Comment:
Composite keys are rarely used. In addition to that, if bake requires additional conventions to use them, almost all people cannot use.
I think that when people want to use composite keys, they have some unavoidable reasons. For example, they must use 3rd party's tables and cannot change the schema. But this implementation never helps them :(
It would better to use real table constraints if you want to support composite keys.